strapi/codemods
# cli/use-separate-commands

The purpose of this PR is to finalize the CLI and improve its UX.  

## What does it do
- Use separate commands instead of options flag
- Make `formatCode` recursively format files in a folder
- Fix issues with jscodeshift transforms
- Update the README
- Checks for a clean git repo when running transforms

## Important files to review
- `codemods/bin/commands`
- `codemods/bin/cli.js`
- `codemods/lib/global/utils/is-path-strapi-app.js`
- `codemods/lib/global/utils/is-clean-git-repo.js`
- `codemods/lib/global/utils/promp-user.js`

## How to test

From the root of the directory `npm install -g .`
Then you can run `strapi-codemods` to launch the CLI

- You can run any command outlined in the README
- Try running transforms in a project without a git repo and expect an error
- Try running transforms in a project without a clean git history and expect an error
